release/ray_release/test_automation/state_machine.py

```diff
@@ -28,20 +39,44 @@ def __init__(self, test: Test) -> None:
             TestState.JAILED: TestState.PASSING
             if self._jailed_to_passing()
             else TestState.JAILED,
-            # passing -> failing
+            # passing -> failing OR passing -> consistently_failing
             TestState.PASSING: TestState.FAILING
             if self._passing_to_failing()
-            else TestState.PASSING,
-            # failing -> passing OR failing -> jailed
+            else (
+                TestState.CONSITENTLY_FAILING
+                if self._passing_to_consistently_failing()
+                else TestState.PASSING
+            ),
+            # failing -> passing OR failing -> consistently_failing
             TestState.FAILING: TestState.PASSING
             if self._failing_to_passing()
-            else (TestState.JAILED if self._failing_to_jailed() else TestState.FAILING),
+            else (
+                TestState.CONSITENTLY_FAILING
+                if self._failing_to_consistently_failing()
+                else TestState.FAILING
+            ),
+            # consistently_failing -> passing OR
+            # consistently_failing -> jailed
+            TestState.CONSITENTLY_FAILING: TestState.PASSING
+            if self._consistently_failing_to_passing()
+            else (
+                TestState.JAILED
+                if self._consistently_failing_to_jailed()
+                else TestState.CONSITENTLY_FAILING
+            ),
         }
         if not self.ray_repo:
-            github_token = TestStateMachine.github_token = boto3.client(
-                "secretsmanager", region_name="us-west-2"
-            ).get_secret_value(SecretId=AWS_SECRET_GITHUB)["SecretString"]
+            github_token = self._get_secret_token(AWS_SECRET_GITHUB)
             self.ray_repo = Github(github_token).get_repo(RAY_REPO)
+        if not self.ray_buildkite:
+            buildkite_token = self._get_secret_token(AWS_SECRET_BUILDKITE)
+            self.ray_buildkite = Buildkite()
+            self.ray_buildkite.set_access_token(buildkite_token)
+
+    def _get_secret_token(self, secret_id: str) -> str:
+        return boto3.client("secretsmanager", region_name="us-west-2").get_secret_value(
+            SecretId=secret_id
+        )["SecretString"]
```

Comment:
this function does not have anything to do with `self`. so maybe, extract this out of the class as a stand-alone helper function?